make/migrations/postgresql/0030_2.0.0_schema.up.sql

```diff
@@ -184,3 +184,43 @@ CREATE TABLE audit_log
  username       varchar(255) NOT NULL,
  op_time        timestamp default CURRENT_TIMESTAMP
 );
+
+/*migrate access log to audit log*/
+/*TODO drop table access_log?*/
+DO $$
+DECLARE
+    access RECORD;
+BEGIN
+    FOR access IN SELECT * FROM access_log
+    LOOP
+      /*insert project create and delete*/
+      IF access.operation = 'create' OR access.operation = 'delete' THEN
+          INSERT INTO audit_log (project_id, operation, resource_type, resource, username, op_time) VALUES (access.project_id, access.operation, 'project', access.repo_name, access.username, access.op_time);
+      END IF;
+      /*insert push image -> create artifact*/
+      IF access.operation = 'push' THEN
+          INSERT INTO audit_log (project_id, operation, resource_type, resource, username, op_time) VALUES (access.project_id, 'create', 'artifact', CONCAT(access.repo_name,'/',access.repo_tag), access.username, access.op_time);
+      END IF;
+      /*insert pull image -> pull tag*/
+      IF access.operation = 'pull' THEN
+          INSERT INTO audit_log (project_id, operation, resource_type, resource, username, op_time) VALUES (access.project_id, 'pull', 'tag', CONCAT(access.repo_name,'/',access.repo_tag), access.username, access.op_time);
```

Comment:
The resource type should be `artifact`

Reply:
the resource type is `artifact` when pull by digest? If pull by tag, the type shouldn't be `tag`?

Reply:
updated.